src/android/CHIPTool/README.md

```diff
@@ -6,3 +6,63 @@ a CHIP device
 
 -   Connection to the chip device's soft AP will be a manual step until pairing
     is implemented.
+
+===========================================
+
+Steps to build CHIP for Android apps
+
+Pre-conditions: Have Android NDK downloaded to your machine. Set $ANDROID_NDK_HOME environment variable to point to the NDK package is downloaded.
+
+1. Checkout the CHIP repo
+
+2. In commandline / Terminal, 'cd' into the top CHIP directory
+
+3. May have to run 'autoconf' / 'autoreconf', if changes to configure.ac have been made.
+
+4. Run './bootstrap -w make'
+
+5. Exit out of the checked out CHIP repo directory and using 'mkdir [some-dir]' create
+a new empty one where you want the generated output files to go.
+
+6. 'cd' into this newly created directory
+
+7. Export the following variables
+
+ABIs that can be used as TARGET
+    armeabi-v7a:  armv7a-linux-androideabi
+    arm64-v8a:	  arch64-linux-android
+    x86:	        i686-linux-android
+    x86-64:	      x86_64-linux-android
+
+NDK OS Variants:
+    macOS:	        darwin-x86_64
+    Linux:	        linux-x86_64
+    32-bit Windows	windows
+    64-bit Windows	windows-x86_64
+
+export TOOLCHAIN=$ANDROID_NDK_HOME/toolchains/llvm/prebuilt/darwin-x86_64 // Use the corresponding OS variant here.
+export TARGET=aarch64-linux-android // Use desired ABI
+export API=21
+export AR=$TOOLCHAIN/bin/$TARGET-ar
+export AS=$TOOLCHAIN/bin/$TARGET-as
+export CC=$TOOLCHAIN/bin/$TARGET$API-clang
+export CXX=$TOOLCHAIN/bin/$TARGET$API-clang++
+export LD=$TOOLCHAIN/bin/$TARGET-ld
+export RANLIB=$TOOLCHAIN/bin/$TARGET-ranlib
+export STRIP=$TOOLCHAIN/bin/$TARGET-strip
+
+8. Then run '../connectedhomeip/configure --host=$TARGET --with-crypto=mbedtls --enable-tests=no --enable-shared'
+
+9. Finally 'make'
```

Comment:
There's an issue with a Java include path here.

In file included from ../../../../connectedhomeip/src/setup_payload/java/SetupPayloadParser-JNI.cpp:4:
/Library/Java/JavaVirtualMachines/jdk-12.0.1.jdk/Contents/Home/include/jni.h:45:10: fatal error: 'jni_md.h' file not found
#include "jni_md.h"
         ^~~~~~~~~~

Reply:
@vidhis88 ^

Reply:
Updated the README to include:
JNI_INCLUDE_DIRS=$ANDROID_NDK_HOME/sysroot/usr/include
to fix this error.